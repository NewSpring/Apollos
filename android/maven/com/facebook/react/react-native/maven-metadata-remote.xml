<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>28.0.0</release>
    <versions>
      <version>28.0.0</version>
    </versions>
    <lastUpdated>20180702224301</lastUpdated>
  </versioning>
</metadata>
