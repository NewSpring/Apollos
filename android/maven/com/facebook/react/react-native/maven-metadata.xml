<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>25.0.0</release>
    <versions>
      <version>25.0.0</version>
    </versions>
    <lastUpdated>20180118183717</lastUpdated>
  </versioning>
</metadata>
