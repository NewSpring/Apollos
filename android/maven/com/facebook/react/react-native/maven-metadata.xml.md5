4d33011c5cd3744e3f0a6d3462b247db