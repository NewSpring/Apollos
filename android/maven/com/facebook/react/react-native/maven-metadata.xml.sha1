8ec2ecca6c881de314f48a8ab786d36bf4a50ada