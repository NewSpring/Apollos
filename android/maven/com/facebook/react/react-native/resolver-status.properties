#NOTE: This is an internal implementation file, its format can be changed without prior notice.
#Thu Jan 18 10:37:17 PST 2018
maven-metadata-remote.xml.lastUpdated=1516300637918
