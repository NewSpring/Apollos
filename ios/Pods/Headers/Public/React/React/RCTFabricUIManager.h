../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h