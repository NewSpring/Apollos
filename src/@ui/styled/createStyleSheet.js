import { StyleSheet } from 'react-native';
import { flatten } from 'lodash';

export const cachedStyles = {};
export const styleHasher = JSON.stringify; // todo: how bad is this?

// Uses cached or generates a new StyleSheet for a given style prop
const createStyleSheet = (stylesToGenerate) => {
  let styles = flatten([stylesToGenerate]); // Need to make sure we're working with a flat array
  const styleSheet = {}; // passed to StyleSheet.create later

  // Load style from cache or add style to stylesheet
  styles.forEach((style, index) => {
    if (typeof style !== 'object' || !style) return;
    const hash = styleHasher(style);
    if (cachedStyles[hash]) {
      styles[index] = cachedStyles[hash];
    } else {
      styleSheet[`${index}`] = style;
    }
  });

  if (Object.keys(styleSheet).length) {
    // Generate the new stylesheet
    const generatedStyleSheet = StyleSheet.create(styleSheet);

    // Process the generated stylesheet
    Object.keys(generatedStyleSheet).forEach((key) => {
      const index = parseInt(key, 0);
      const generatedStyle = generatedStyleSheet[key];
      const hash = styleHasher(styles[index]);

      // add generated style to cache
      cachedStyles[hash] = generatedStyle;

      // swap generated style into result list
      styles[index] = generatedStyle;
    });
  }

  if (styles.length === 1) styles = styles[0]; // eslint-disable-line
  return styles;
};

export default createStyleSheet;
