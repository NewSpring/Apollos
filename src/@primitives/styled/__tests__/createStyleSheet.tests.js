import React from 'react';

import { View } from 'react-native';
import renderer from 'react-test-renderer';
import createStyleSheet, { cachedStyles } from '../createStyleSheet';

describe('the createStyleSheet function', () => {
  it('returns styles that work on components', () => {
    const style = createStyleSheet({ backgroundColor: 'red' });
    const tree = renderer.create(<View style={style} />);
    expect(tree).toMatchSnapshot();
  });
  it('uses a style cache when using same style twice', () => {
    const style = { backgroundColor: 'red' };
    const a = createStyleSheet(style);
    const b = createStyleSheet(style);
    expect(a).toEqual(b);
    expect(Object.keys(cachedStyles)).toHaveLength(1);
  });
  it('doesnt fail when passed an already generated style', () => {
    const style = createStyleSheet(0); // simulating StyleSheet.create() used elsewhere
    expect(style).toEqual(0);
  });
  it('returns a single style if possible', () => {
    const style = createStyleSheet([{ backgroundColor: 'red' }]);
    expect(Array.isArray(style)).toBeFalsy();
  });
});
