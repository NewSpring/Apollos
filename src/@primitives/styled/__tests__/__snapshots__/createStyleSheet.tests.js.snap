// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the createStyleSheet function returns styles that work on components 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;
